### ISSUE TITLE
Incorrect indent error in multiline array expression

### ISSUE THREAD

**Issue by jurcovicova** [yannickcr/eslint-plugin-react#902]
After upgrade to 6.4.1 following code causes indent error:

``` js
const myFunction = () => (
  [
    <Tag
      {...properties}
    />,
    <Tag
      {...properties}
    />,
    <Tag
      {...properties}
    />,
  ]
)
```

error:

```
6:5   error  Expected indentation of 2 space characters but found 4    react/jsx-indent 
9:5   error  Expected indentation of 2 space characters but found 4    react/jsx-indent 
```

E.g. the error refers to second and third instance of the tag `Tag` in array.

Another case with the same problem:

```
              {condition && [
                  <Tag key="a" onClick={() => {
                    // some code
                  }} />,
                  <Tag key="b" onClick={() => {
                    // some code
                  }} />,
                ]
              }
```

error:

```
5:19  error  Expected indentation of 16 space characters but found 18  react/jsx-indent
```

E.g. the error refers to second instance of the tag `Tag` in array.

**Comment by ljharb** [comment]
The first case is a bug, but the second is correct - the line the starting `[` is on determines the indentation for the rest. In other words, it wants:
```jsx
              {condition && [
                <Tag key="a" onClick={() => {
                  // some code
                }} />,
                <Tag key="b" onClick={() => {
                  // some code
                }} />,
              ]}
```

**Comment by jurcovicova** [comment]
@ljharb In that case, I would expect two errors - one for line 2 and one for line 5. At least, when I donwgrade to 6.3.0 and intentionally indent it wrong, I get one error for each wrongly indented tag.

**Comment by ljharb** [comment]
Yes, good point - it is warning, but not in the right places.

**Comment by bigslycat** [comment]
I confirm. 
![2016-10-12 11-14-12](https://cloud.githubusercontent.com/assets/1121997/19302669/21c367e2-9076-11e6-82f7-a31f341e06d5.png)

**Comment by ljharb** [comment]
@bigslycat on 6.4.1?

**Comment by bigslycat** [comment]
@ljharb oh, sry. =) I check it now — on 6.4.1 it was fixed.

**Comment by nilsml** [comment]
I'm on version 7.5.1 and this is still an issue.

**Comment by ljharb** [comment]
The first case passes on the latest default branch.

The second one passes as well, which is actually correct with two-space indentation.

Please open a new issue with code, config, and expected results, if you think there's still an issue in the latest version of the plugin.
